projects/aca-playwright-shared/src/page-objects/components/dialogs/password-overlay-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Page } from '@playwright/test';
+import { BaseComponent } from '.././base.component';
+import { timeouts } from '../../../utils';
+
+export class PasswordOverlayDialogComponent extends BaseComponent {
+  private static rootElement = '.cdk-overlay-pane';
+
+  public closeButton = this.getChild('[data-automation-id="adf-password-dialog-close"]');
+  public submitButton = this.getChild('[data-automation-id="adf-password-dialog-submit"]');
+  public passwordInput = this.getChild('[data-automation-id="adf-password-dialog-input"]');
+  public errorMessage = this.getChild('[data-automation-id="adf-password-dialog-error"]');
+
+  constructor(page: Page, rootElement = PasswordOverlayDialogComponent.rootElement) {
+    super(page, rootElement);
+  }
+
+  async waitForDialogToOpen(): Promise<void> {
+    await this.passwordInput.waitFor({ state: 'visible', timeout: timeouts.short });
+    await this.passwordInput.isVisible();
+  }
+
+  async waitForDialogToClose(): Promise<void> {
+    await this.passwordInput.waitFor({ state: 'detached', timeout: timeouts.large });
+  }
+
+  async isDialogOpen(): Promise<boolean> {
+    try {
+      await this.waitForDialogToOpen();
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  async isCloseEnabled(): Promise<boolean> {
```

Comment:
name does not match the behaviour, it says "enabled" but checks for "visibile"

Reply:
rename the method.